packages/playwright/src/runner/testServer.ts

```diff
@@ -252,7 +252,7 @@ export class TestServerDispatcher implements TestServerInterface {
     config.cliListOnly = true;
 
     const status = await runTasks(new TestRun(config, reporter), [
-      createLoadTask('out-of-process', { failOnLoadErrors: false, filterOnly: false }),
+      createLoadTask('out-of-process', { failOnLoadErrors: false, filterOnly: false, populateDependencies: true }),
```

Comment:
We probably want an option for this, since most invocations of `listTests()` do not really need to populate dependencies, and it's costly to run vite build. I'd put an option into `initialize()`. Perhaps UI mode will opt into that at some point as well.

Reply:
We're only running the Vite build if there's a component testing plugin registered, in which case we need to run Vite to get a correct result. In the common case without component testing, `populateDependencies: true` is a no-op. Could you outline a scenario where this adds unnecessary cost?

Reply:
Wait, ignore me - listing tests does *not* need a vite build to return a correct result. Dependency population is what needs the vite build, but one could be interested in listing tests without caring about dependency population. Gonna implement your recommendation.

Reply:
done in https://github.com/microsoft/playwright/pull/32583/commits/7827ca77c547838385b0eca71cdf52e2dad514d1